### ISSUE TITLE
Parsing error: Computed property names are not allowed in enums.

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/vue-eslint-parser#274]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I'm using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue).
- [x] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo and open the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo if there is no solution.
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.34.0

### What version of `eslint-plugin-vue` and `vue-eslint-parser` are you using?

- vue-eslint-parser@10.2.0
- eslint-plugin-vue@10.4.0

### What did you do?

see https://github.com/vuejs/eslint-plugin-vue/actions/runs/17426870338/job/49476333521?pr=2920 and https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/fixtures/script-indent/ts-enum-member-01.vue:

<details>
<summary>Configuration</summary>

```json
{"languageOptions": {"parserOptions": {"parser":"@typescript-eslint/parser"}}}
```
</details>

```vue
<script lang="ts">
enum Direction {
  // comment
  Up = 1,
  // comment
  Down,
  Left
    = 2,
  [
    Right
  ] =
    3,
}
</script>
```

### What did you expect to happen?

This test case used to pass with https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.41.0, but started to fail when `@typescript-eslint/parser` was updated to [v8.42.0](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.42.0).

### What actually happened?

`Parsing error: Computed property names are not allowed in enums.`

### Link to Minimal Reproducible Example

https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/fixtures/script-indent/ts-enum-member-01.vue
